spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_stdp.py

```diff
@@ -289,8 +289,6 @@ def get_n_words_for_plastic_connections(self, n_connections):
         :rtype: int
         """
         synapse_structure = self.__timing_dependence.synaptic_structure
-        if self.__pad_to_length is not None:
-            n_connections = max(n_connections, self.__pad_to_length)
```

Comment:
I don't know if we can remove this from here completely though; this fixes the current error, but may result in under-allocation of synaptic memory (it may be worth running integration tests to see what happens), since this method is called from elsewhere.

I think there needs to be an internal method that this method and get_max_synapses both call, where this public method then adds the pad_to_length check.

Reply:
Possibly, yes.  Will try running the integration tests again now and see if it flags anything.

Reply:
As you suspected, it has flagged a problem.  Will fix...